testtools/deferredruntest.py

```diff
@@ -232,8 +233,10 @@ def _run_core(self):
             [error_observer.gotEvent, full_observer.emit],
             self._blocking_run_deferred, spinner)
 
+        full_log.seek(0, os.SEEK_SET)
+        log_content = full_log.readlines()
         self.case.addDetail(
-            'twisted-log', Content(UTF8_TEXT, full_log.readlines))
+            'twisted-log', Content(UTF8_TEXT, lambda: log_content))
```

Comment:
I don't see the need for the lambda here? Why not just keep this as it was:

 Content(UTF8_TEXT, full_log.readlines)